packages/black-forest-labs/src/bfl-image-model.ts

```diff
@@ -0,0 +1,280 @@
+import type {
+  ImageModelV2,
+  ImageModelV2CallWarning,
+} from '@ai-sdk/provider';
+import type { Resolvable } from '@ai-sdk/provider-utils';
+import {
+  FetchFunction,
+  combineHeaders,
+  createBinaryResponseHandler,
+  createJsonErrorResponseHandler,
+  createJsonResponseHandler,
+  createStatusCodeErrorResponseHandler,
+  getFromApi,
+  postJsonToApi,
+  resolve,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod/v4';
+import type { BlackForestLabsAspectRatio } from './bfl-image-settings';
+import { BlackForestLabsImageModelId } from './bfl-image-settings';
+
+interface BlackForestLabsImageModelConfig {
+  provider: string;
+  baseURL: string;
+  headers?: Resolvable<Record<string, string | undefined>>;
+  fetch?: FetchFunction;
+  _internal?: {
+    currentDate?: () => Date;
+    pollIntervalMs?: number;
+  };
+}
+
+export class BlackForestLabsImageModel implements ImageModelV2 {
+  readonly specificationVersion = 'v2';
+  readonly maxImagesPerCall = 1;
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  constructor(
+    readonly modelId: BlackForestLabsImageModelId,
+    private readonly config: BlackForestLabsImageModelConfig,
+  ) {}
+
+  async doGenerate({
+    prompt,
+    size,
+    aspectRatio,
+    providerOptions,
+    headers,
+    abortSignal,
+  }: Parameters<ImageModelV2['doGenerate']>[0]): Promise<
+    Awaited<ReturnType<ImageModelV2['doGenerate']>>
+  > {
+    const warnings: Array<ImageModelV2CallWarning> = [];
+
+    const finalAspectRatio =
+      aspectRatio ?? (size ? convertSizeToAspectRatio(size) : undefined);
+
+    if (size && !aspectRatio) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'size',
+        details:
+          'Black Forest Labs does not accept width/height. Deriving aspect_ratio from size.',
+      });
+    } else if (size && aspectRatio) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'size',
+        details:
+          'Black Forest Labs ignores size when aspectRatio is provided.',
+      });
+    }
+
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+
+    const submit = await postJsonToApi({
+      url: `${this.config.baseURL}/${this.modelId}`,
+      headers: combineHeaders(await resolve(this.config.headers), headers),
+      body: {
+        prompt,
+        ...(finalAspectRatio ? { aspect_ratio: finalAspectRatio } : {}),
+        ...(providerOptions.bfl ?? {}),
+        ...(providerOptions['black-forest-labs'] ?? {}),
+      },
+      failedResponseHandler: bflFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(bflSubmitSchema),
+      abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    const pollUrl = submit.value.polling_url;
```

Comment:
is there a synchronous api or only polling? asking as AI SDK prefers synchronous when available.